src/plugins/data/common/search/expressions/esql.ts

```diff
@@ -9,22 +9,32 @@
 import type { KibanaRequest } from '@kbn/core/server';
 import { esFieldTypeToKibanaFieldType } from '@kbn/field-types';
 import { i18n } from '@kbn/i18n';
-import type { IKibanaSearchResponse, IKibanaSearchRequest } from '@kbn/search-types';
+import type {
+  IKibanaSearchRequest,
+  IKibanaSearchResponse,
+  ISearchGeneric,
+} from '@kbn/search-types';
 import type { Datatable, ExpressionFunctionDefinition } from '@kbn/expressions-plugin/common';
 import { RequestAdapter } from '@kbn/inspector-plugin/common';
 import { getStartEndParams } from '@kbn/esql-utils';
-
 import { zipObject } from 'lodash';
-import { Observable, defer, throwError } from 'rxjs';
-import { catchError, map, switchMap, tap } from 'rxjs';
-import { buildEsQuery } from '@kbn/es-query';
-import type { ISearchGeneric } from '@kbn/search-types';
-import type { ESQLSearchResponse, ESQLSearchParams } from '@kbn/es-types';
+import { catchError, defer, map, Observable, switchMap, tap, throwError } from 'rxjs';
+import { buildEsQuery, type Filter } from '@kbn/es-query';
+import type { ESQLSearchParams, ESQLSearchResponse } from '@kbn/es-types';
 import { getEsQueryConfig } from '../../es_query';
 import { getTime } from '../../query';
-import { ESQL_ASYNC_SEARCH_STRATEGY, KibanaContext, ESQL_TABLE_TYPE } from '..';
+import { ESQL_ASYNC_SEARCH_STRATEGY, ESQL_TABLE_TYPE, KibanaContext } from '..';
```

Comment:
Most of this is just import optimization, the only actual change here is adding the `type Filter` to the import from `@kbn/es-query`.